10-kubernetes-inputs.tf

```diff
@@ -96,3 +96,72 @@ variable "azure_policy_enabled" {
   description = "Enable the Azure Policy addon"
   default     = false
 }
+
+variable "node_os_maintenance_window_duration" {
+  type = number
+
+  description = "Duration of maintenance window in hours"
+  default     = 4
+
+  validation {
+    condition     = var.node_os_maintenance_window_duration >= 4
+    error_message = "Maintenance window duration must be at least 4 hours when node_os_channel_upgrade is enabled."
+  }
+}
+
+variable "node_os_maintenance_window_frequency" {
+  type = string
+
+  description = "Frequency of maintenance window, Daily or Weekly"
+  default     = "Daily"
+
+  validation {
+    condition     = var.node_os_maintenance_window_frequency == "Daily" || var.node_os_maintenance_window_frequency == "Weekly"
```

Comment:
Match this to the `node_os_maintenance_window_day_of_week` setup, `contains` is expandable versus an OR statement.

Reply:
Only seeing this now, will try and implement that. thanks Marty